File: backend/src/main/java/com/chami/backend/services/NotificationService.java
package com.chami.backend.services;

import com.chami.backend.dtos.NotificationDTO;
import com.chami.backend.models.Notification.NotificationType;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface NotificationService {
    
    // Create a notification
    NotificationDTO createNotification(Long senderId, Long receiverId, NotificationType type, 
                                       String message, Long relatedEntityId, String relatedEntityType);
    
    // Get all notifications for current user
    Page<NotificationDTO> getUserNotifications(Long userId, Pageable pageable);
    
    // Get unread notification count
    Long getUnreadNotificationCount(Long userId);
    
    // Mark a notification as read
    NotificationDTO markAsRead(Long notificationId);
    
    // Mark all notifications as read for a user
    int markAllAsRead(Long userId);
    
    // Delete a notification
    void deleteNotification(Long notificationId);
    
    // Create like notification
    NotificationDTO createLikeNotification(Long likerId, Long contentOwnerId, Long contentId, String contentType);
    
    // Create comment notification
    NotificationDTO createCommentNotification(Long commenterId, Long contentOwnerId, Long contentId, String contentType);
    
    // Create enrollment notification
    NotificationDTO createEnrollmentNotification(Long enrollerId, Long planOwnerId, Long planId);
}

File: frontend/src/services/notification.service.js
import api from './api';

class NotificationService {
  // Get user's notifications with pagination
  getNotifications(page = 0, size = 10) {
    return api.get(`/notifications?page=${page}&size=${size}`);
  }
  
  // Get count of unread notifications
  getUnreadCount() {
    return api.get('/notifications/unread-count');
  }
  
  // Mark a notification as read
  markAsRead(notificationId) {
    return api.patch(`/notifications/${notificationId}/read`);
  }
  
  // Mark all notifications as read
  markAllAsRead() {
    return api.patch('/notifications/mark-all-read');
  }
  
  // Delete a notification
  deleteNotification(notificationId) {
    return api.delete(`/notifications/${notificationId}`);
  }
}

export default new NotificationService();
